### PR TITLE
Bump controller-tools

### PR DESCRIPTION
**Pull request by anyasabo**
Fix https://github.com/elastic/cloud-on-k8s/issues/2490

Controller tools 0.2.5 includes k8s 1.17 deps, and also includes some of the new openapi properties to support new CRD features. This causes a few issues:
- `x-kubernetes-int-or-string` seems to cause the 1.15 api server to interpret the schema as structural, and so is sad that there is no `type: object` property (which we intentionally remove for Openshift 3.11 support). Removing the `x-kubernetes-int-or-string` property allows the API server to accept the manifests on both 1.11 and 1.15, though kubectl validation fails so you need to pass it `--validation=false`.

- Kubectl validation fails because there are other new properties that are added (`x-kubernetes-list-type` and `x-kubernetes-list-map-keys`). I removed these properties too to make applying the CRDs easier, even though they are not strictly required by the API server.

I believe these properties will be required in k8s 1.19 when v1beta1 CRD support is removed and structural schema is required, but that is a bit away. All of this stems from supporting 1.11 which has a bug that requires us to remove `type:object` from the root. Once we stop supporting 1.11 this all becomes much simpler.

Other options than the approach in this PR that may be worthwhile:
- Add a flag to crd-gen to disable adding these markers, though since it is only an issue for <=1.11 support I am not sure how much support it would get
- Pin an older version of controller-tools (looks like 0.2.2 is the latest we could use just browsing release notes)

Also added the namespace to some unit tests that were passing but don't seem like they should have been in older versions (and were failing now).

It's also not clear to me if we should target v1.0.1 with this.

### CODE REVIEW

File: pkg/controller/elasticsearch/driver/upgrade_test.go [hunk 1]
```diff
@@ -36,11 +36,11 @@ func Test_podsToUpgrade(t *testing.T) {
 			args: args{
 				statefulSets: sset.StatefulSetList{
 					sset.TestSset{
-						Name: "masters", Replicas: 2, Master: true,
+						Name: "masters", Namespace: TestEsNamespace, Replicas: 2, Master: true,
```
Comment (sebgl) [inline pkg/controller/elasticsearch/driver/upgrade_test.go]: I'm surprised to see that in the PR.
Does it fail if not set?

Comment (anyasabo) [reply pkg/controller/elasticsearch/driver/upgrade_test.go]: Yeah, it was surprising to me that this ever passed. The NS was added in 
9d3aaa8 so if you check out any of the ones before that you should be able to see.

**Comment by sebgl** [comment]
We may want to wait for https://github.com/kubernetes-sigs/controller-tools/issues/392 to be resolved first?

**Comment by anyasabo** [comment]
Ah I missed that issue somehow, yes that would definitely make the kustomize part of this PR simpler. I could see this going either way though -- we could merge this as is then use the flag once it is available, or we could just close this and wait for the flag.

File: config/crds/patches/apm-kibana-patches.yaml [hunk 2]
```diff
@@ -21,3 +21,13 @@
 # that would maybe not match the user's k8s version.
 - op: remove
   path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties
+
+# without this the API server complains with "The CustomResourceDefinition "elasticsearches.elasticsearch.k8s.elastic.co" is invalid: spec.validation.openAPIV3Schema.type: Required value: must not be empty at the root" on 1.15 at least. When we stop supporting 1.11 we should remove these as well.
```
Comment (sebgl) [inline config/crds/patches/apm-kibana-patches.yaml]: We should include something like
```
# TODO: remove once https://github.com/kubernetes-sigs/controller-tools/issues/392 is fixed
```
IMO this is a bug in controller-tools.
Also it looks like this errors out in 1.14 as well, so we probably want to remove the 1.11 mention.

Comment (sebgl) [reply config/crds/patches/apm-kibana-patches.yaml]: Nit: can you reformat the comments to multiple lines so they read better on a small-ish screen?

**Comment by charith-elastic** [comment]
Jenkins test this please

**Comment by anyasabo** [comment]
This is pending https://github.com/elastic/cloud-on-k8s/pull/2611 as there are some oddities in the license generator on OSX vs Linux that should be resolved in that PR. Right now the CI check is failing for the notice file

**Pull request merged**